doc/http-api/backend_swagger.yml

```diff
@@ -208,7 +196,50 @@ paths:
           required: true
           type: string
         - name: limit
+          in: query
+          description: An upper bound on the number of messages to fetch from the archive
+          required: true
+          type: integer
+      description: Retrieve recent message from archive.
+      responses:
+        200:
+          description: The messages archived for the given user and the other party
+          schema:
+            title: records
+            type: array
+            items:
+              title: record
+              type: object
+              properties:
+                sender:
+                  type: string
+                  format: JID
+                timestamp:
+                  type: integer
+                message_id:
+                  type: string
+                body:
+                  type: string
+  /messages/{caller}/{other}/{before}:
+    get:
+      parameters:
+        - name: caller
           in: path
+          description: The user archive in question
+          required: true
+          type: string
+        - name: other
+          in: path
+          description: The other party
+          required: true
+          type: string
+        - name: before
```

Comment:
It's a matter of taste, but I think this might be better off as an QS parameter with default value _now_.